test/Dialect/Seq/firreg.mlir

```diff
@@ -248,6 +248,7 @@ hw.module private @InitReg1(%clock: i1, %reset: i1, %io_d: i32, %io_en: i1) -> (
   // CHECK-NEXT:     } else { 
   // CHECK-NEXT:       sv.passign %reg, %6 : i32 
   // CHECK-NEXT:     }
+  // CHECK-NEXT:     sv.passign %reg3, %2 : i32
```

Comment:
Just adding this test is equivalent to checking the behavior of this patch. I opted to also add the explicit test at the bottom as I am concerned that our future selves will not be able to grok the importance of this line otherwise.